var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Variorum(Package):
+    """Variorum is a library providing vendor-neutral interfaces for
+       monitoring and controlling underlying hardware features.
+    """
+
+    homepage = "https://variorum.readthedocs.io"
+    git = "https://github.com/llnl/variorum.git"
+    url = "https://github.com/LLNL/variorum/archive/v0.1.0.tar.gz"
+
+    maintainers = ["slabasan", "rountree"]
+
+    version("0.1.0", tag="v0.1.0")
+
+    ############
+    # Variants #
+    ############
+    variant("shared", default=True, description="Build Variorum as shared lib")
+    variant("test", default=True, description="Enable Variorum unit tests")
+
+    # variants for language support
+    variant("python", default=True, description="Build Variorum Python support")
+
+    # variants for dev-tools (docs, etc)
+    variant("docs", default=False, description="Build Variorum's documentation")
+    variant("log", default=False, description="Enable Variorum's logs")
+    variant("debug", default=False, description="Enable Variorum's debug")
+
+    ########################
+    # Package dependencies #
+    ########################
+    depends_on("cmake@2.8:", type="build")
+    depends_on("hwloc@1.11.9", when="@:0.5.1+hwloc")
+
+    #########################
+    # Documentation related #
+    #########################
+    depends_on("py-sphinx", when="+python+docs", type="build")
+
+    root_cmakelists_dir = "../src"
+
+    def install(self, spec, prefix):
```

Comment:
If you switch to `CMakePackage` base class above, you can get rid of this whole method and only define `cmake_args(self)`.